### PR TITLE
Fix buffer size computation and log partial sends in Server

### PR DESCRIPTION
**Pull request by Hagb**
Similar to #3.

### CODE REVIEW

File: src/Server/Connection.cpp [hunk 1]
```diff
@@ -117,7 +119,7 @@ void Connection::send(const void *packet, size_t size)
 	std::cout << "] " << size << " bytes: " << reinterpret_cast<const Lobbies::Packet *>(packet)->toString() << std::endl;
 	logMutex.unlock();
 #endif
-	this->_socket->send(packet, size, sent);
+	this->_socket->send(packet, size);
```
Comment (Hagb) [inline src/Server/Connection.cpp]: Now I know that this modification is useless.

**Review by PinkySmile — APPROVED** [review]
Seems fine to me, is it ready to be merged ?

**Comment by Hagb** [comment]
> Seems fine to me, is it ready to be merged ?

yes.
(squash please)

**Comment by Hagb** [comment]
@Gegel85  wait... i have a concern in fact

**Comment by Hagb** [comment]
In this PR, `send` are changed to be logged _after_ it is executed. would it make debugging more hard or more confusing in a production environment, if these are some problems (for example a freed socket is used) in `send` causing the program to exit unexpectedly?

**Comment by Hagb** [comment]
A core dump would be helpful in such a case. In CN lobby a similar problem happened, but they forgot to enable core dump...

**Comment by Hagb** [comment]
> In this PR, `send` are changed to be logged _after_ it is executed. would it make debugging more hard or more confusing in a production environment, if these are some problems (for example a freed socket is used) in `send` causing the program to exit unexpectedly?

After thinking more, I realize that it is not such a concern, for `_socket->getRemoteAddress()` and `_socket->getRemotePort()` are called before the log is really printed. So I think the PR is ready to be merged.

**Pull request merged**